src/replay_stage.rs

```diff
@@ -180,222 +71,169 @@ impl ReplayStage {
-
-                    if current_slot.is_none() {
-                        let new_slot = Self::get_next_slot(
-                            &blocktree,
-                            prev_slot.expect("prev_slot must exist"),
-                        );
-                        if new_slot.is_some() {
-                            trace!("{} replay_stage: new_slot found: {:?}", my_id, new_slot);
-                            // Reset the state
-                            bank = Self::create_and_set_working_bank(
-                                new_slot.unwrap(),
-                                &bank_forks,
+                    Self::generate_new_bank_forks(&blocktree, &mut bank_forks.write().unwrap());
+                    let live_bank_ids = bank_forks.read().unwrap().active_banks();
+                    trace!("live banks {:?}", live_bank_ids);
+                    let mut votable: Vec<u64> = vec![];
+                    for bank_id in live_bank_ids {
+                        let bank = bank_forks.read().unwrap().get(bank_id).unwrap().clone();
+                        if !Self::is_tpu(&bank, my_id) {
+                            Self::replay_blocktree_into_bank(
                                 &bank,
-                            );
-                            current_slot = new_slot;
-                            Self::reset_state(
-                                bank.ticks_per_slot(),
-                                current_slot.unwrap(),
-                                &mut max_tick_height_for_slot,
-                                &mut current_blob_index,
-                            );
-                        } else {
-                            continue;
+                                &blocktree,
+                                &mut progress,
+                                &forward_entry_sender,
+                            )?;
                         }
-                    }
-
-                    // current_slot must be Some(x) by this point
-                    let slot = current_slot.unwrap();
-
-                    // Fetch the next entries from the database
-                    let entries = {
-                        if current_leader_id != my_id {
-                            info!(
-                                "{} replay_stage: asking for entries from slot: {}, bi: {}",
-                                my_id, slot, current_blob_index
-                            );
-                            if let Ok(entries) = blocktree.get_slot_entries(
-                                slot,
-                                current_blob_index,
-                                Some(MAX_ENTRY_RECV_PER_ITER as u64),
-                            ) {
-                                entries
-                            } else {
-                                vec![]
+                        let max_tick_height = (bank_id + 1) * bank.ticks_per_slot() - 1;
+                        if bank.tick_height() == max_tick_height {
```

Comment:
if it's > max_tick_height, we do what?

Reply:
that means its a bad fork and will eventually get timed out/pruned?